### ISSUE TITLE
Something strange with LoadSyntheaTables

### ISSUE THREAD

**Issue by mgabetta** [OHDSI/ETL-Synthea#80]
Hi!
I'm experiencing a very strange issue with the LoadSyntheaTables step.
Attached two Synthea sets:

- "set_3000" generated with "./run_synthea -p 3000 -c /root/synthea/config/synthea.custom.properties -m lung_cancer"
- "set_100" generated with "./run_synthea -p 100 -c /root/synthea/config/synthea.custom.properties -m lung_cancer"

[set_3000.zip](https://github.com/OHDSI/ETL-Synthea/files/5944715/set_3000.zip)
[set_100.zip](https://github.com/OHDSI/ETL-Synthea/files/5944729/set_100.zip)

When launching LoadSyntheaTables, set_100 is correctly managed, while set_3000 produces "conditions" and "procedures" tables with messy CODE (see attached screenshots).
 
Wrong codes in set_3000
![Schermata 2021-02-08 alle 16 48 41](https://user-images.githubusercontent.com/5736000/107243690-8d340000-6a2d-11eb-9aef-c662285eabde.png)

Right codes n set_100
![Schermata 2021-02-08 alle 16 55 32](https://user-images.githubusercontent.com/5736000/107244676-940f4280-6a2e-11eb-956e-26008458c542.png)

Do you have any ideas?

**Comment by chrisroederucdenver** [comment]
Are all the codes in set_3000 mangled? What do they look like in the original Synthea files?

**Comment by mgabetta** [comment]
Hi Chris!

Here's a distinct of all codes...all mangled:
![Schermata 2021-02-08 alle 17 58 23](https://user-images.githubusercontent.com/5736000/107254118-4814cb80-6a37-11eb-96e5-5cf71901b2aa.png)

Here's an extract of the original Synthea (complete version in the zip attached to my first message):
![Schermata 2021-02-08 alle 18 00 04](https://user-images.githubusercontent.com/5736000/107254346-801c0e80-6a37-11eb-8096-5e752b58b9c7.png)

**Comment by chrisroederucdenver** [comment]
“LoadSyntheaTables step” I missed that.

When I’ve done that, I’ve let PostgreSQL infer the schema. What database are you using, where do the table definitions come from, what are they?
I’d look my own up, but it’s a few steps away in a cloud db.

**Comment by mgabetta** [comment]
I'm using Postgres and the tables definitions of Synthea come from the ETLSyntheaBuilder::CreateSyntheaTables step.
Output of Postgres version() command:

` PostgreSQL 10.15 on x86_64-pc-linux-gnu, compiled by gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-39), 64-bit`

**Comment by AnthonyMolinaro** [comment]
@mgabetta  Thanks for your continued worked on this!

You call those codes messy, some would say they are nice and neat (after all they are all zero).  ; )

I ran synthea with 3000 people specifying the lung cancel module, but wasn't able to reproduce what you're seeing.  Can you share your custom config file?

**Comment by mgabetta** [comment]
Here's the config file (zipped because of Github):

[synthea.custom.properties.zip](https://github.com/OHDSI/ETL-Synthea/files/5949694/synthea.custom.properties.zip)

So if you start from the csv inside set_3000.zip (attached to my first message), the conditions table is correctly populated?

**Comment by chrisroederucdenver** [comment]
It looks like it. Depending what you use to load a CSV, the double commas for a null value might need modified.
The table definitions for the synthea tables are in inst/sql/sql_server/create_synthea_tables.sql and there you can see the fields are mostly character type, so it should load. To narrow this down, you could look at https://www.postgresql.org/docs/10/sql-copy.html to see how to use postgres commands to load the csv. There’s 99.9% chance this will work, but it will give you some confidence in the data, your postgresql install etc.  If you are using the bulk-load script, this is what it uses. Note the section on NULL.

If you are not using bulk-load, I assume LoadSyntheaTables.r? Are you familiar with R?

**Comment by mgabetta** [comment]
I'm using R (`ETLSyntheaBuilder::LoadSyntheaTables(connectionDetails = cd, syntheaSchema = syntheaSchema, syntheaFileLoc = syntheaFileLoc)`) for populating Synthea tables...I was trying to stay within this project boundaries to move a Synthea run in OMOP.

Now, since the Synthea step produces those mangled codes, I've solved by importing each Synthea csv inside its DB tables with my database client (Dbeaver) and then coming back to ETLSyntheaBuilder with the LoadVocabFromCsv step.

Nonetheless I'm still curious about that error :)

**Comment by chrisroederucdenver** [comment]
All good. It’s good to hear you’re making progress.

I’m curious too and may have a look later in the week.
-Chris

**Comment by AnthonyMolinaro** [comment]
@chrisroederucdenver, @mgabetta  Thanks for your work here guys. 

@mgabetta I tried it again using your config file and couldn't reproduce the error.  To be safe and based on @chrisroederucdenver 's comments, I added the na.strings param to the last commit so that empty fields (either missing or with a single blank space) are considered NA in R and thus converted to NULL in SQL (as fix as the one applied to loadVocabFromCsv).

I also noticed that in your custom config file you have exporter.symptoms.mode = 1, while the default has exporter.symptoms.mode = 0.  I tested both ways and it didn't make a difference on my end.  If after pulling the latest update you still have the error, perhaps try changing the 1 to 0.
